resources/custom-resources/customize-cognito-ui/src/main/java/com/amazon/aws/partners/saasfactory/saasboost/CustomizeCognitoUi.java

```diff
@@ -134,6 +137,25 @@ public Object handleRequest(Map<String, Object> event, Context context) {
         return null;
     }
 
+    protected void waitForActiveDomain(String userPoolDomain) throws Exception {
+        // Cognito cannot set UI customization until UserPoolDomain is ACTIVE
+        // ACTIVE, CREATING, DELETING, FAILED, UPDATING, UNKNOWN_TO_SDK_VERSION
+        DomainStatusType domainStatus = DomainStatusType.CREATING;
+        List<DomainStatusType> terminalStatuses = List.of(
+                DomainStatusType.ACTIVE,
+                DomainStatusType.FAILED,
+                DomainStatusType.UNKNOWN_TO_SDK_VERSION);
+        do {
+            domainStatus = cognito.describeUserPoolDomain(request -> request.domain(userPoolDomain))
```

Comment:
Will we be throttled here in a way that the SDK client won't be able to deal with?

Reply:
Good point, I'll implement some throttling backoff.